modules/build/src/main/scala/scala/build/Inputs.scala

```diff
@@ -237,6 +236,8 @@ object Inputs {
           Inputs.ScalaFile(d.path, p.subRelativeTo(d.path))
         case p if p.last.endsWith(".sc") =>
           Inputs.Script(d.path, p.subRelativeTo(d.path))
+        case p if p.last.endsWith(".c") || p.last.endsWith(".h") =>
```

Comment:
this will include `.c`/`.h` files as `Inputs` regardless if we're in Scala native mode, no?
we should have an extra check here to only include those for Scala native builds

Reply:
That's a good point, but OTOH, users might be enabling / disabling Scala Native on-the-fly in watch mode by using a `//> using platform "native"`. So we should accept them upfront anyway IMO, and they'll get picked or not later on.

Reply:
yeah, that makes sense.
however, when Scala native is disabled and any `.c`/`.h` input are passed, we should at least show a warning that they're being ignored, IMO